client/src/components/ProTx/components/util.js

```diff
@@ -282,48 +317,67 @@ export const getFipsIdName = currentGeoid => {
  * @param {*} typesDataArray
  * @returns
  */
-export const getBarVertTrace = (traceY, traceX, traceName, traceFillColor) => {
+export const getBarTrace = (
+  traceY,
+  traceX,
+  traceName,
+  traceFillColor,
+  barOrientation
+) => {
+  let xData;
+  let yData;
+
+  if (barOrientation === 'v') {
+    xData = traceX;
+    yData = traceY;
+  }
+
+  if (barOrientation === 'h') {
+    xData = traceY;
+    yData = traceX;
+  }
+
   return {
-    y: [traceY],
-    x: [traceX],
+    y: [yData],
+    x: [xData],
     name: traceName,
     type: 'bar',
-    orientation: 'v',
+    orientation: barOrientation,
     marker: {
       line: {
         color: ['#111111'],
-        width: 1
+        width: 0.1
       },
       color: [traceFillColor]
     }
   };
 };
 
-/**
- * Assign an imported color theme for use in plot generation.
- */
-export const plotColors = THEME_CB12_MAIN;
+export const getTraceFillColor = (targetPlot, catcode, unique = false) => {
+  let barColorIndex = 10;
+  let barColor = histColors[barColorIndex];
 
-/**
- * Define array of category codes.
- */
-export const categoryCodes = [
-  'ABAN',
-  'EMAB',
-  'LBTR',
-  'MDNG',
-  'NSUP',
-  'PHAB',
-  'PHNG',
-  'RAPR',
-  'SXAB',
-  'SXTR',
-  'NA'
-];
-
-export const getCategoryColorDestructured = catcode => {
-  const indexKey = categoryCodes.indexOf(catcode);
-  const barColor = plotColors[indexKey];
+  if (targetPlot === 'maltreatment') {
+    const indexKey = categoryCodes.indexOf(catcode);
+    barColor = plotColors[indexKey];
+    return barColor;
+  }
+
+  if (targetPlot === 'observed') {
+    if (unique) {
+      barColorIndex = 10;
```

Comment:
never used.

Reply:
Ha! This was bad logic, *should* have been used but wasnt being. It is now. =)